src/os/bluestore/KernelDevice.cc

```diff
@@ -683,17 +683,41 @@ int KernelDevice::_sync_write(uint64_t off, bufferlist &bl, bool buffered)
   }
   vector<iovec> iov;
   bl.prepare_iov(&iov);
-  int r = ::pwritev(buffered ? fd_buffered : fd_direct,
-		    &iov[0], iov.size(), off);
 
-  if (r < 0) {
-    r = -errno;
-    derr << __func__ << " pwritev error: " << cpp_strerror(r) << dendl;
-    return r;
-  }
+  auto left = len;
+  auto o = off;
+  size_t idx = 0;
+  do {
+    auto r = ::pwritev(buffered ? fd_buffered : fd_direct,
+      &iov[idx], iov.size() - idx, o);
+
+    if (r < 0) {
+      r = -errno;
+      derr << __func__ << " pwritev error: " << cpp_strerror(r) << dendl;
+      return r;
+    }
+    o += r;
+    left -= r;
+    if (left) {
```

Comment:
In paranoid mode, what if r > left ?

Reply:
Mind leaving as-is for now? This is a backport PR which makes no sense to fix separately.  While fixing all the releases looks a bit overkill